src/main/kotlin/no/skatteetaten/aurora/boober/feature/FluentbitSidecarFeature.kt

```diff
@@ -0,0 +1,354 @@
+import com.fkorotkov.kubernetes.resources
+import com.fkorotkov.kubernetes.secretKeyRef
+import io.fabric8.kubernetes.api.model.Container
+import io.fabric8.kubernetes.api.model.Quantity
+import io.fabric8.kubernetes.api.model.Volume
+import io.fabric8.openshift.api.model.DeploymentConfig
+import no.skatteetaten.aurora.boober.model.AuroraConfigFieldHandler
+import no.skatteetaten.aurora.boober.model.AuroraContextCommand
+import no.skatteetaten.aurora.boober.model.AuroraDeploymentSpec
+import no.skatteetaten.aurora.boober.model.AuroraResource
+import no.skatteetaten.aurora.boober.model.findSubKeys
+import no.skatteetaten.aurora.boober.utils.addIfNotNull
+import org.apache.commons.codec.binary.Base64
+import org.springframework.beans.factory.annotation.Value
+
+val AuroraDeploymentSpec.loggingIndex: String? get() = this.getOrNull<String>("logging/index")
+val AuroraDeploymentSpec.fluentConfigName: String? get() = "${this.name}-fluent-config"
+val AuroraDeploymentSpec.fluentParserName: String? get() = "${this.name}-fluent-parser"
+val AuroraDeploymentSpec.hecSecretName: String? get() = "${this.name}-hec"
+val hecTokenKey: String = "HEC_TOKEN"
+val splunkHostKey: String = "SPLUNK_HOST"
+val splunkPortKey: String = "SPLUNK_PORT"
+
+val logApplication: String = "application"
+val logAuditText: String = "audit_text"
+val logAuditJson: String = "audit_json"
+val logSlow: String = "slow"
+val logGC: String = "gc"
+val logSensitive: String = "sensitive"
+val logStacktrace: String = "stacktrace"
+val logAccess: String = "access"
+val knownLogs: Set<String> = setOf(logApplication, logAuditText, logAuditJson, logSlow, logGC, logSensitive, logStacktrace, logAccess)
+
+val parserMountPath = "/fluent-bit/parser"
+val parsersFileName = "parsers.conf"
+
+/*
+Fluentbit sidecar feature provisions fluentd as sidecar with fluent bit configuration based on aurora config.
+ */
+@org.springframework.stereotype.Service
+class FluentbitSidecarFeature(
+    @Value("\${splunk.hec.token}") val hecToken: String,
+    @Value("\${splunk.hec.url}") val splunkUrl: String,
+    @Value("\${splunk.hec.port}") val splunkPort: String
+) : Feature {
+    override fun enable(header: AuroraDeploymentSpec): Boolean {
+        // TODO validate logic
+        return header.type in listOf(
+                TemplateType.deploy,
+                TemplateType.development
+        )
+    }
+
+    override fun handlers(header: AuroraDeploymentSpec, cmd: AuroraContextCommand): Set<AuroraConfigFieldHandler> {
+        val customHandlers = cmd.applicationFiles.findSubKeys("logging/custom").flatMap { key ->
+            listOf(
+                AuroraConfigFieldHandler(
+                    "logging/custom/$key/index",
+                    defaultValue = header.loggingIndex
+                ),
+                AuroraConfigFieldHandler(
+                    name = "logging/custom/$key/pattern"
+                    // validator =
+                    // TODO validate that this is requered
```

Comment:
Det skal være en egen opsjon for å sette noe til required

Reply:
Skal se hva jeg kan finne og fikse

Reply:
validator = { it.required("<beskjed>") }),